test/matrix/kernelmatrix.jl

```diff
@@ -31,16 +31,29 @@ KernelFunctions.kappa(::ToySimpleKernel, d) = exp(-d / 2)
 
             @test kernelmatrix(k, x, y) ≈ transpose(kernelmatrix(k, y, x))
 
-            @test kerneldiagmatrix(k, x) ≈ diag(kernelmatrix(k, x))
+            @test kernelmatrix_diag(k, x) ≈ diag(kernelmatrix(k, x))
 
             tmp = Matrix{Float64}(undef, length(x), length(y))
             @test kernelmatrix!(tmp, k, x, y) ≈ kernelmatrix(k, x, y)
+            @test tmp ≈ kernelmatrix(k, x, y)
 
             tmp_square = Matrix{Float64}(undef, length(x), length(x))
             @test kernelmatrix!(tmp_square, k, x) ≈ kernelmatrix(k, x)
+            @test tmp_square ≈ kernelmatrix(k, x)
 
             tmp_diag = Vector{Float64}(undef, length(x))
-            @test kerneldiagmatrix!(tmp_diag, k, x) ≈ kerneldiagmatrix(k, x)
+            @test kernelmatrix_diag!(tmp_diag, k, x) ≈ kernelmatrix_diag(k, x)
+            @test tmp_diag ≈ kernelmatrix_diag(k, x)
+
+            # Test deprecations
+            @test @test_deprecated kerneldiagmatrix(k, x) == kernelmatrix_diag(k, x)
+            @test @test_deprecated kerneldiagmatrix(k, x, x) == kernelmatrix_diag(k, x, x)
+
+            tmp_diag_dep = Vector{Float64}(undef, length(x))
+            @test_deprecated kerneldiagmatrix!(tmp_diag_dep, k, x)
+            @test tmp_diag_dep == kernelmatrix_diag(k, x)
+            @test_deprecated kerneldiagmatrix!(tmp_diag_dep, k, x, x)
```

Comment:
It would be cleaner/safer to reset `tmp_diag_dep`. Otherwise the test of the return value might pass even if the function is implemented incorrectly:

```suggestion

            tmp_diag_dep = Vector{Float64}(undef, length(x))            
            @test_deprecated kerneldiagmatrix!(tmp_diag_dep, k, x, x)
```

BTW it seems we only test `kernelmatrix_diag!(out, k, x)` but not `kernelmatrix_diag!(out, k, x, y)` above in the non-deprecated section?

Reply:
I'll add that as well - though I was already wondering about what the use-case for kernelmatrix_diag(k, x, y) is! it seems like code that needs maintaining without any benefit I can see from it - what would you use it for?